src/server.c

```diff
@@ -4006,21 +4006,6 @@ int processCommand(client *c) {
             rejectCommandSds(c, err);
```

Comment:
Wouldn't your argument also extend to these two checks? You can check if the server has passwords enabled by sending both an invalid command or a command that doesn't exist.

```
127.0.0.1:6379> set Foo
(error) ERR wrong number of arguments for 'set' command
127.0.0.1:6379> set FOO BAR
(error) NOAUTH Authentication required.
127.0.0.1:6379> bah
(error) ERR unknown command 'bah', with args beginning with: 
```

Seems like we should cover them all.

Reply:
ok, I have thought of this too, and think this check can be put in advance, but it makes sense, I will cover them together.

Reply:
btw, i am going to handle this as well.
```
            /* Handle possible security attacks. */
            if (!strcasecmp(c->argv[0]->ptr, "host:") || !strcasecmp(c->argv[0]->ptr, "post")) {
                securityWarningCommand(c);
                return C_ERR;
            }
```

Reply:
On the other hand, this can let people check whether the server supports xxx command

Reply:
I believe `AUTH` is primarily intended to protect user data. I understand the consistency concern regarding returning the same error message when AUTH is validated, but I don’t see a security issue with revealing a command's presence, arity, or protected status?

BTW, this is a behavior change, though perhaps not a breaking one.

Reply:
> I believe AUTH is primarily intended to protect user data

i think it should protect everything according to the existing code.

> I understand the consistency concern regarding returning the same error message when AUTH is validated, but I don’t see a security issue with revealing a command's presence, arity, or protected status?

yes, it is not a security issue, i just think it's better this way

> BTW, this is a behavior change, though perhaps not a breaking one.

yes, it is a behavior change but i guess it is safe.